### ISSUE TITLE
User feedback: Improve keyboard accessibility

### ISSUE THREAD

**Issue by karmatosed** [WordPress/gutenberg#2801]
This report has come in from a user testing:

"The editor is rather inaccessible. I am blind, so use a screen reader to read my computer. I tested with orca on linux so far, and it... failed pretty miserably. While tabbing around the interface, buttons would seem to be there then not there at random intervals, and some controls I could access (for example the disabled move buttons when you only have one block) suddenly disappeared entirely upon adding another block. In addition, keyboard shortcuts are *badly* needed, not just for blind/accessibility use.

Please, please don't go live with this plugin in core until it is fully accessible. I have really been looking for an accessible page builder for the longest time, but (usually because the page builders are comercial products) the developers cannot devote time to a miniscule number of possible users. This would be the chance to shine for gutenberg -- make it accessible to the blind and more. I really can't imagine what I would do if WordPress's default (only?) editor suddenly became so inaccessible, but it would not be a happy time, as I rely on WordPress for all of my sites, and suddenly being unable to modify and publish content would be a *huge* blow."

**Comment by karmatosed** [comment]
More feedback from another user:

"Well I have a lot of issue with all those button controls.
They work but no one will know what B I img Ol, ul, etc are, I do because I did web programming class, also the img button when clicked really should put you to a field where you upload an image.
I did just paste the link in the document but I may have hit the button twice there is no feadback when the button was pressed.
On the flipside, for the blind this function won't matter as such, we don't usually input images at least I don't.
We do incert links to audio and maybe things in the main ariea and would like them read as such.
For the blind it may be usefull to have some cloud ocr package or something render the image as well a descriptive one just so we know what we are viewing if you insist on having us use images perio"

**Comment by karmatosed** [comment]
"I can't use any of the controls to the side of blocks that appear on mouseover (Move Up, Move Down, Show Inspector, Delete). Not being able to delete a block (with it being so easy to add new ones) means that I'll never use this editor. There's something about the way Dragon works that issuing any command makes those icons disappear (even if i physically bring them up by mousing over a block) so I can't even inspect them."

**Comment by afercia** [comment]
Just to start with, please see

Keyboard interaction: standardize the blocks behavior #2031
Unexpected reverse-tab behaviour #1934
Image block keyboard focus and tab order #1829
Improve the block multi-selection and keyboard interaction #1308

**Comment by karmatosed** [comment]
Closing as we have issues for this.
